oneview_redfish_toolkit/tests/blueprints/test_resource_block.py

```diff
@@ -0,0 +1,218 @@
+
+        with open(
+            'oneview_redfish_toolkit/mockups/redfish'
+            '/ServerHardwareResourceBlock.json'
+        ) as f:
+            expected_resource_block = json.load(f)
+
+        g.oneview_client.index_resources.get_all.return_value = [
+            {"category": "server-hardware"}
+        ]
+        g.oneview_client.server_hardware.get.return_value = server_hardware
+        g.oneview_client.server_profile_templates.get_all.return_value = \
+            server_profile_templates
+
+        response = self.app.get(
+            "/redfish/v1/CompositionService/ResourceBlocks"
+            "/30303437-3034-4D32-3230-313133364752")
+
+        result = json.loads(response.data.decode("utf-8"))
+
+        self.assertEqual(status.HTTP_200_OK, response.status_code)
+        self.assertEqual("application/json", response.mimetype)
+        self.assertEqual(expected_resource_block, result)
+
+    @mock.patch.object(resource_block, 'g')
+    def test_get_resource_block_computer_system_not_found(self, g):
+        error = HPOneViewException({
+            'errorCode': 'RESOURCE_NOT_FOUND',
+            'message': 'server-hardware not found'
+        })
+        g.oneview_client.server_hardware.get.side_effect = error
+
+        response = self.app.get(
+            "/redfish/v1/CompositionService/ResourceBlocks"
+            "/30303437-3034-4D32-3230-313133364752/Systems/2M201136GR")
+
+        self.assertEqual(status.HTTP_404_NOT_FOUND, response.status_code)
+        self.assertEqual("application/json", response.mimetype)
+
+    @mock.patch.object(resource_block, 'g')
+    def test_get_computer_system_not_found(self, g):
+        error = HPOneViewException({
+            'errorCode': 'RESOURCE_NOT_FOUND',
+            'message': 'server-hardware not found'
+        })
+        g.oneview_client.server_hardware.get.side_effect = error
+
+        response = self.app.get(
+            "/redfish/v1/CompositionService/ResourceBlocks"
+            "/30303437-3034-4D32-3230-313133364752/Systems/2M201136GR")
+
+        self.assertEqual(status.HTTP_404_NOT_FOUND, response.status_code)
+        self.assertEqual("application/json", response.mimetype)
+
+    @mock.patch.object(resource_block, 'g')
+    def test_get_computer_system_invalid_serial(self, g):
+
+        with open(
+            'oneview_redfish_toolkit/mockups/oneview/ServerHardware.json'
+        ) as f:
+            server_hardware = json.load(f)
+
+        g.oneview_client.server_hardware.get.return_value = server_hardware
+
+        response = self.app.get(
+            "/redfish/v1/CompositionService/ResourceBlocks"
+            "/30303437-3034-4D32-3230-313133364752/Systems/1234567890")
+
+        self.assertEqual(status.HTTP_404_NOT_FOUND, response.status_code)
+        self.assertEqual("application/json", response.mimetype)
+
+    @mock.patch.object(resource_block, 'g')
+    def test_get_computer_system(self, g):
+        with open(
```

Comment:
This block is used by more than one test, right?! We could put it inside the setUp or out of the test class (but in the same file) to reuse it.

Reply:
done

Reply:
Since it's not used in all test cases, place it on setUp will make it runs this block for every test case. Maybe the better way it's to place it on setUpClass, but in order to use setUpClass on children classes from TestBase, we need to refactor TestBase class. For this refactoring maybe we can use setUpModule to place the code that today is on the setUp from TestBase.